### PR TITLE
Hotel Eligibility Toggle fix-#2156

### PR DESCRIPTION
**Pull request by migetman9**
#2156. This fix will add a toggle checkbox to registration/form that allows an attendee to be eligible for hotel space. Currently requested is the ability to make sure only a STOPS-level user toggles this box. need something like `{% if c.STOPS in session.admin.restrictions %}`

### CODE REVIEW

File: uber/templates/accounts/sitemap.html [hunk 1]
```diff
@@ -6,10 +6,12 @@
 
 {% for section, pagelist in pages %}
     <div class="panel panel-default">
-    <div class="panel-heading">{{ section|title }}</div>
+    <div class="panel-heading"><a href="../{{ section }}/index">{{ section|title }}</a></div>
         <ul class="nav nav-pills">
         {% for page in pagelist %}
+            {% if page.name != 'Index' %}
```
Comment (kitsuta) [inline uber/templates/accounts/sitemap.html]: Unfortunately, because this PR includes commits from another PR (it's really tricky to remember to avoid that), you'll want to rebase this PR off of master in order to get rid of these lines. If you just remove this set of lines in another commit, it'll cause a merge conflict.

File: uber/templates/regform.html [hunk 2]
```diff
@@ -367,6 +367,22 @@ <h4 class="h4-inline">
     </div>
 </div>
 
+{% if c.PAGE_PATH == '/registration/form' %}
+<div class="form-group staffing">
+    <label for="rooming" class="col-sm-2 optional-field control-label">Can Room?</label>
+    <div class="col-sm-6">
+        {% checkbox attendee.hotel_eligible %}
+        This attendee
+        {% if attendee.hotel_eligible %}
```
Comment (kitsuta) [inline uber/templates/regform.html]: I get the idea behind this, but I think it'll end up being confusing for admins.

The checkbox is always ticked if the variable is true, and not ticked if the variable is false. What will happen here is that, if an admin views an attendee's page and they are not hotel eligible, it will say "This attendee cannot request room space." next to an _unticked_ checkmark. The admin may think that the checkmark has to be ticked in order to deny room space to the attendee. This, of course, isn't true.

TLDR, the checkbox already communicates the attendee's current hotel eligibility status, so it should always say "This attendee can request room space."

**Comment by kitsuta** [comment]
I just realized this was filed against the core plugin.

The plugin system is quite highly dependent on the fact that plugins keep to themselves. The hierarchy is:
1. The core plugin, which must always work entirely on its own with no errors.
2. Feature plugins (e.g., panels, hotel), which can depend on stuff in the core plugin but not on stuff in other feature plugins.
3. Event plugins (e.g., magfest, magprime), which can safely assume that certain feature plugins will be installed.

Adding a variable from the `hotel` plugin to a core template violates this hierarchy and will crash the Django template for anyone who installs the core plugin without the `hotel` plugin. This is, unfortunately, unacceptable.

As pointed out in https://github.com/magfest/ubersystem/issues/2156, this can be added via baseextra.html in one of the 'event' plugins we use. As also explained in #2156, we can only have one `baseextra.html` in the system once everything is installed. There's already one extant in our `magclassic` repo, so I'd recommend putting this in our `magprime` repo. Remember that you can reference regextra.html to see how we use jQuery to position these 'injected' fields.

File: uber/templates/accounts/sitemap.html (hunk 1 above)
Comment (kitsuta) [inline uber/templates/accounts/sitemap.html]: Since this change will have to be effectively redone, this won't be an issue. Which is good cause this stuff is super annoying to do. D:
